Implement prolog server

ljos opened the pull request:
Implemented and commented on how to use prolog for extreme_startup.

I added some convenience predicates so people won't have to figure out those things themselves and can just start hacking in prolog. I added parsing of string lists and number lists, as well as formatting of those types of lists. I also show an example (hopefully not one that is used) for how to parse strings.

ljos committed:
Implement prolog server

olillevik commented:
Thank you for your contribution. I would love to have a skeleton of a prolog-server present. Some minor modification is needed before accepting the pull request.

A README.txt containing instructions for use similar to the java jetty example. Comments for the file answers.pl explaining the reason for the files existence. Based on the comments for each method I assume the file should not be present at all. The other servers presented are skeletons and contain very few helper methods besides extracting variables from the http-request.

ljos commented:
I updated the PR and removed the extra convenience functions. I followed the jetty example. The remaining code in `answers.pl` is needed to show how definite clause grammars work.  Definite clause grammars are important to show since there aren't regular expressions in prolog.

olillevik commented:
Thank you for the contribution! May your server do well in upcoming competitions.
